pkg/resources/connection.go

```diff
@@ -0,0 +1,313 @@
+		MinItems:    1,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the connection.",
+	},
+	ShowOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `SHOW CONNECTIONS` for the given secret.",
+		Elem: &schema.Resource{
+			Schema: schemas.ShowConnectionSchema,
+		},
+	},
+	FullyQualifiedNameAttributeName: schemas.FullyQualifiedNameSchema,
+}
+
+func Connection() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateContextConnection,
+		ReadContext:   ReadContextConnection(true),
+		UpdateContext: UpdateContextConnection,
+		DeleteContext: DeleteContextConnection,
+		Description:   "Resource used to manage connections. For more information, check [connection documentation](https://docs.snowflake.com/en/sql-reference/sql/create-connection.html).",
+		Schema:        connectionSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportName[sdk.AccountObjectIdentifier],
+		},
+	}
+}
+
+func CreateContextConnection(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	id, err := sdk.ParseAccountObjectIdentifier(d.Get("name").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	request := sdk.NewCreateConnectionRequest(id)
+
+	if v, ok := d.GetOk("as_replica_of"); ok {
+		externalObjectId, err := sdk.ParseExternalObjectIdentifier(v.(string))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		request.WithAsReplicaOf(externalObjectId)
+	}
+
+	if v, ok := d.GetOk("comment"); ok {
+		request.WithComment(v.(string))
+	}
+
+	err = client.Connections.Create(ctx, request)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(helpers.EncodeResourceIdentifier(id))
+
+	if v, ok := d.GetOk("is_primary"); ok && v.(bool) {
+		err := client.Connections.Alter(ctx, sdk.NewAlterConnectionRequest(id).
+			WithPrimary(v.(bool)),
+		)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	if v, ok := d.GetOk("enable_failover_to_accounts"); ok {
+		enableFailoverConfig := v.([]any)
+
+		enableFailoverToAccountsList := make([]sdk.AccountIdentifier, 0)
+		for _, enableToAccount := range enableFailoverConfig {
+			accountInConfig := enableToAccount.(string)
+			accountIdentifier := sdk.NewAccountIdentifierFromFullyQualifiedName(accountInConfig)
+
+			enableFailoverToAccountsList = append(enableFailoverToAccountsList, accountIdentifier)
+		}
+
+		err := client.Connections.Alter(ctx, sdk.NewAlterConnectionRequest(id).
+			WithEnableConnectionFailover(*sdk.NewEnableConnectionFailoverRequest().
```

Comment:
It regards the sdk, but shouldn't the account list be required in `NewEnableConnectionFailoverRequest`? Having empty `ENABLE FAILOVER TO ACCOUNTS` doesn't make sense.

Reply:
Unless this is the way to clean it?

Reply:
to clean it there is a `alter connection <conn_name> disable failover`